check_hwgroup.py

```diff
@@ -106,147 +143,57 @@ def probe(self):
-                                'activeOn'
-                            )[int(inpAlarmSetup)]
-                        ),
-                        float(inpValue)
-                    )
-
-                (
-                    outValue,
-                    outName,
-                    outType,
-                    outMode
-                ) = [self.SNMPReq(
-                    '{}.3.{}.2.1.{}.{}'.format(
-                        enterprise,
-                        {
-                            'Damocles': 4,
-                            'Poseidon': 3
-                        }[self.deviceType],
-                        OID,
-                        self.output
-                    )
-                ) for OID in range(2, 6)]
-
-                return (
-                    '{} [Type: {}, Mode: {}]'.format(
-                        outName,
-                        (
-                            'relay (off, on)',
-                            'rts (-10V,+10V)',
-                            'dtr (0V,+10V)'
-                        )[int(outType)],
-                        (
-                            'manual',
-                            'autoAlarm',
-                            'autoTriggerEq',
-                            'autoTriggerHi',
-                            'autoTriggerLo'
-                        )[int(outMode)]
-                    ),
-                    float(outValue)
-                )
-            except KeyError:
-                raise self.notSupported # pylint: disable=raise-missing-from
+                sensorID = int(self.SNMPReq('{}.{}.{}'.format(enterprise, sensor_ids[self.deviceType], ident)))
+            except (ValueError, CheckHWGroupError):
+                continue
+            if sensorID == self.sensor:
+                sensorID = ident
+                break
+        else:
+            raise CheckHWGroupError('Sensor ID ({}) not found'.format(self.sensor))
+
+        (sensName, sensState, sensValue) = [self.SNMPReq('{}.{}.{}'.format(enterprise, OID, sensorID)) for OID in sensor_paths[self.deviceType]]
+
+        if not int(sensState):
+            raise CheckHWGroupError('getting sensor values failed')
+
+        return (sensName, float(sensValue) / 10)
+
+    def _probe_contact(self):
+        (inpValue, inpName, inpAlarmSetup, inpAlarmState) = [self.SNMPReq('{}.3.{}.1.1.{}.{}'.format(enterprise,{'Damocles': 4, 'Poseidon': 3}[self.deviceType], OID, self.contact)) for OID in range(2, 6)]
 
+        contactName = '{} [AlarmState: {}, AlarmSetup: {}]'.format(
+            inpName,
+            ('normal', 'alarm')[int(inpAlarmState)],
+            ('inactive', 'activeOff', 'activeOn')[int(inpAlarmSetup)])
+
+        return (contactName, float(inpValue))
+
+    def _probe_output(self):
+        (outValue, outName, outType, outMode) = [self.SNMPReq('{}.3.{}.2.1.{}.{}'.format(enterprise, {'Damocles': 4, 'Poseidon': 3}[self.deviceType], OID, self.output)) for OID in range(2, 6)]
```

Comment:
should this dict not also be extracted into a variable like `sensor_paths`?

Reply:
Yeah, I thought about that too but decided against it since it's so small. But for consistency we probably should.

Reply:
Added a variable for this